common/utils/jwt.go

```diff
@@ -25,6 +25,10 @@ func ExtractFieldFromJWT(secret string, token string, field string) ([]string, e
 	}
 
 	if claims, ok := jwt_token.Claims.(jwt.MapClaims); ok && jwt_token.Valid {
+		if claims["exp"] == nil {
```

Comment:
Cool, you can do statement shenanigans. Since I wanted to use the exp returned later, I wasn't able to do that because of scoping. Still nice to know.

Are we talkin about the same line 32 expired time.Now() check? It seems to work for me, and there's an E2E test that validates it. Am I missing something?

I did add the type check into the flow though: 347e6ea

Reply:
> Are we talkin about the same line 32 expired time.Now() check? It seems to work for me, and there's an E2E test that validates it. Am I missing something?

https://github.com/HackIllinois/api/blob/9609de0a1fb18cfef643fb71a07fecbe1e439a2f/common/utils/jwt.go#L37-L39
I'm actually unsure if line 38 is reachable. In the [JWT docs](https://pkg.go.dev/github.com/golang-jwt/jwt/v4#Parse), it says `jwt.Parse()` does in fact validate not only the signature but also any claims attached to the token by default (e.g. if a `exp` is on the token, it is checked during this stage).

> Edit: I did have to keep the nil check as well, otherwise you get a "nil is not an interface" which isn't very descriptive. (https://github.com/HackIllinois/api/commit/46e064a4b30aad656ea17ed7a1988e62f59971d3)

Can you add the steps to recreate this bug? ` _, ok := claims["exp"].(float64)` should, in fact, catch `nil`s as well (regardless if the key was set or not). Check out the linked playground and see tests 2 and 5: https://go.dev/play/p/CjR1n-gTnRU

Reply:
Yeah, this is my bad. I was just doing it with the literal `nil`, which isn't typed as an interface so of course it fails. 

I see what you mean now about the expired token. Looks like our check was redundant.

However, we do still have to check for the existence of the field, as if it doesn't exist it won't check for it.

Resolved by 618e388.